engine/crates/runtime/src/hooks.rs

```diff
@@ -248,27 +248,92 @@ impl<'a> ExecutedSubgraphRequest<'a> {
     }
 }
 
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone)]
 pub struct Operation<'a> {
-    pub name: Option<&'a str>,
+    pub name: Option<String>,
```

Comment:
See the other comment on lifetime issues. Would be great to have this as `Option<&'a str>`, but it just doesn't compile easily with that. Maybe `&'a Option<String>` could work, but that also feels a bit sus.

Reply:
won't work, you have no choice but top copy that one